AEPIdentity/Sources/IdentityMap.swift

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright 2020 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+import Foundation
+
+enum XDMAuthenticationState: String, Codable {
+    case ambiguous
+    case authenticated
+    case loggedOut
+
+    static func authStateFromMobileAuthState(authState: MobileVisitorAuthenticationState?) -> XDMAuthenticationState {
+        switch authState {
+        case .authenticated:
+            return .authenticated
+        case .loggedOut:
+            return .loggedOut
+        default:
+            return .ambiguous
+        }
+    }
+}
+
+/// Defines a map containing a set of end user identities, keyed on either namespace integration code or the namespace ID of the identity.
+/// Within each namespace, the identity is unique. The values of the map are an array, meaning that more than one identity of each namespace may be carried.
+struct IdentityMap: Equatable {
+    private var items: [String: [IdentityItem]] = [:]
+
+    /// Adds an `IdentityItem` to this map. If an item is added which shares the same `namespace` and `id` as an item
+    /// already in the map, then the new item replaces the existing item.
+    /// - Parameters:
+    ///   - namespace: The namespace for this identity
+    ///   - id: Identity of the consumer in the related namespace.
+    ///   - authenticationState: The state this identity is authenticated as for this observed ExperienceEvent.
+    ///   - primary: Indicates if this identity is the preferred identity. It is used as a hint to help systems better organize how identities are queried.
+    mutating func addItem(namespace: String,
+                          id: String,
+                          authenticationState: MobileVisitorAuthenticationState? = nil,
```

Comment:
I suggest to keep these concepts separated, and use XDMAuthenticationState here, this way we can use this API for pure xdm objects and maybe have some helper functions for conversions.